packages/store-sync/src/stash/createStorageAdapter.ts

```diff
@@ -66,10 +69,7 @@ export function createStorageAdapter({ stash }: CreateStorageAdapter): StorageAd
 
         updates.push((pendingRecords[id] = { table, key, value }));
       } else if (log.eventName === "Store_SpliceDynamicData") {
-        const previousValue = pendingRecords[id]
-          ? ({ ...pendingRecords[id].key, ...pendingRecords[id].value } as TableRecord)
-          : getRecord({ stash, table, key });
```

Comment:
we were missing the logic for checking that `pendingRecords[id].value` is defined in the `SpliceDynamicData` branch